### PR TITLE
2.3 bionic netplan bridging vlans and bonds 1764735

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

This adds support for Netplan to parse a lot more of the values that can be set. It adds support for parsing VLAN and Bond information, and ultimately will support creating bridges on those devices.

## QA steps

```
$ juju bootstrap MAASWITHVLANORBOND --bootstrap-series bionic
$ juju deploy cs;~jameinel/ubuntu-lite -m controller --to lxd:0
```

## Documentation changes

Adds support for Bionic, but doesn't otherwise change how a user would interact with the system.

## Bug reference

[lp:1764735](https://bugs.launchpad.net/juju/+bug/1764735)
[lp:1771120](https://bugs.launchpad.net/juju/+bug/1771120)

### CODE REVIEW

**Comment by jameinel** [comment]
[1764735 Bionic Netplan Bonds](https://trello.com/c/LxCNv6e5/215-1764735-bionic-netplan-bonds)

File: network/netplan/netplan.go [hunk 1]
```diff
@@ -76,41 +119,168 @@ type Netplan struct {
...
+		return errors.Trace(err)
+	}
+	np.createBridgeFromInterface(bridgeName, deviceId, &vlan.Interface)
+	np.Network.VLANs[deviceId] = vlan
+	return nil
+}
+
+// BridgeBondById takes a deviceId and creates a bridge with this device
+// using this devices config
+func (np *Netplan) BridgeBondById(deviceId string, bridgeName string) (err error) {
```
Comment (manadart) [inline network/netplan/netplan.go]: I might be tempted to create an abstraction for these 3, but perhaps passing in a device type string and a map pointer makes it less cogent...

Comment (jameinel) [reply network/netplan/netplan.go]: I pulled everything that wasn't type specific into helper functions. I couldn't think of a better way to simplify them, though I would consider switching away from individual BridgeFoo and just have a BridgeDevice that knows about all three types.
Thoughts?

**Comment by jameinel** [comment]
$$merge$$

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**